benchmark/sample_taxa.py

```diff
@@ -0,0 +1,122 @@
+                 |          \---|    \----species 1.4
+                 |              |
+        ---------|              |     /---species 1.5
+                 |               \---|
+                 |                    \---species 1.6
+                 |
+                 |          /-------------species 2.1
+                 |         |
+                  \-----genus 2       /---species 2.3
+                           |     /---|
+                            \---|     \---species 2.4
+                                |
+                                 \--------species 2.5
+
+    The current status is a simple prototype. It indiscriminatingly samples all
+    taxa indicated by subject proteins in the DIAMOND hit table (i.e., all
+    genomes sharing more or less some homology with the query genome).
+
+    In the future this code may require heavy and repeated improvement based on
+    statistical and empirical studies of benchmarking results.
+    """
+    # column names in standard BLAST tabular format
+    m8cols = ['qseqid', 'sseqid', 'pident', 'length', 'mismatch', 'gapopen',
+              'qstart', 'qend', 'sstart', 'send', 'evalue', 'bitscore']
+    df = pd.read_table(hit_table_fp, index_col=None, names=m8cols)
+    # get all subject protein IDs mentioned in the hit table
+    prots = set(df['sseqid'])
+    # get IDs of taxa hosting subject proteins, based on a dictionary
+    taxa = set()
+    with open(prot2tax_dict_fp, 'r') as f:
+        for line in f:
+            # dictionary format: protein ID <tab> taxon IDs separated by comma
+            prot, taxon_list = line.rstrip().split('\t')
+            if prot in prots:
+                taxa.update(taxon_list.split(','))
+    # write selected taxon IDs
+    with open(output_taxa_fp, 'w') as f:
+        for taxon in sorted(taxa):
+            f.write('%s\n' % taxon)
+
+
+@click.command()
+@click.option('--hit-table-fp', required=True,
+              type=click.Path(resolve_path=True, readable=True, exists=True,
+                              file_okay=True),
+              help='Sequence similarity search hit table in standard tabular '
+                   'format (e.g., -outfmt 6 for BLAST or m8 for DIAMOND).')
+@click.option('--prot2tax-dict-fp', required=True,
+              type=click.Path(resolve_path=True, readable=True, exists=True,
+                              file_okay=True),
+              help='Protein to taxon (genome) dictionary file.')
+@click.option('--output-taxa-fp', required=True,
+              type=click.Path(resolve_path=True, readable=True, exists=False,
+                              file_okay=True),
+              help='Output list of selected taxa.')
+def _main(hit_table_fp,
+          prot2tax_dict_fp,
+          output_taxa_fp):
+    """ Select taxa for subsequent analyses
+    """
+    sample_taxa(hit_table_fp,
+                prot2tax_dict_fp,
+                output_taxa_fp)
+
+
+if __name__ == "__main__":
+    _main()
```

Comment:
This script looks good.
I'd maybe change the function name from `_main()` to something more descriptive.
Also, is there a specific reason why you're writing `taxa` to file in `sample_taxa()`, instead of returning it and then writing it to file in `main()`.
In my example, if you just call the function, you get a data structure and if you call the script, you get a file.